tests/jenkins/TestReleaseTagJob.groovy

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+import jenkins.tests.BuildPipelineTest
+import org.junit.Before
+import org.junit.Test
+
+class TestReleaseTagJob extends BuildPipelineTest {
+
+    @Before
+    void setUp() {
+
+        def buildManifest = 'tests/data/opensearch-build-1.1.0.yml'
+
+        def destPath = "${this.workspace}/${buildManifest}"
+
+        def artifactsPath = 'distribution-build-opensearch/1.1.0/123/linux/x64/builds/opensearch/manifest.yml'
+
+        def bucketName = 'job-s3-bucket-name'
+
+        this.registerLibTester(new DownloadFromS3LibTester(destPath, bucketName, artifactsPath, true))
+
+        this.registerLibTester(new CreateReleaseTagLibTester(buildManifest, '1.1.0'))
+
+        super.setUp()
+
+        // Variables for Release Tag Job
+        binding.setVariable('VERSION', '1.1.0')
+        binding.setVariable('BUILD_ID', '123')
+        binding.setVariable('ARTIFACT_BUCKET_NAME', bucketName)
+        binding.setVariable('BUILD_MANIFEST', buildManifest)
+
+        helper.registerAllowedMethod("checkout", [Map], {})
+        helper.registerAllowedMethod("dir", [Map], {})
+        helper.addShMock("git ls-remote --tags https://github.com/opensearch-project/OpenSearch.git 1.1.0 | awk 'NR==1{print \$1}'") { script ->
+            return [stdout: "", exitValue: 0]
+        }
+        helper.addShMock("git ls-remote --tags https://github.com/opensearch-project/common-utils.git 1.1.0.0 | awk 'NR==1{print \$1}'") { script ->
```

Comment:
Why do we want to mock these sh values, since they don't return anything? These mocks would simply be ignored in case this is not found.

Maybe assertions in the lib tester that these sh mocks exists is a better way to go.

Reply:
It returns the commit ID. I mocked different cases for when the tag is not existed so they don't return anything..

Reply:
shMock returning empty strings will not serve any purpose here.

Reply:
These should be a part of libtester since they are specific to the library

Reply:
Had a discussion with @abhinavGupta16 , and new changes are updated.